### ISSUE TITLE
Signing with Trezor is broken

### ISSUE THREAD

**Issue by dennisreimann** [btcpayserver/BTCPayServer.Vault#38]
**Describe the bug**

Importing a Trezor wallet using the Vault works, but signing a transaction does not. 

**Your  BTCPay Environment (please complete the following information):**

- BTCPay Server Version: 1.2.1

- BTCPay Server Vault app version:    2.0.1

- Deployment Method: Dev Env

**Your local environment (please complete the following information):**

- Your operating system: macOS
- Browser: Firefox

**Your hardware wallet details  (please complete the following information):**

- Hardware wallet name: Trezor
- Hardware wallet version: Trezor Bridge 2.0.27
- Hardware wallet firmware, bootloader, microcontroller verisons (where applicable): 1.9.3 and 1.10.2 (latest)

**Additional context**

Original bug report by @pavlenex, maybe he can provide additional information. I confirmed the issue. 

**Screenshots / Video / GIf (if applicable)**

1.9.3 Firmware

https://user-images.githubusercontent.com/886/129888232-c3b93f7d-6d9d-4ccf-90f8-b2a84400859f.MOV

1.10.2 Firmware

https://user-images.githubusercontent.com/886/129889421-1c2809e6-516a-47fa-8374-59826e34a226.MOV

**Comment by pavlenex** [comment]
cc @NicolasDorier

**Comment by NicolasDorier** [comment]
checking

**Comment by NicolasDorier** [comment]
the new release of hwi doesn't see to solve any trezor related stuff... that's weird.

**Comment by mikeyyyv** [comment]
I'm having the exact same issue here. Thank you.

**Comment by NicolasDorier** [comment]
@mikeyyyv trying to find out what happened.

**Comment by NicolasDorier** [comment]
Does the Trezor T has this issue? Sadly I don't have a trezor 1...
Also @mikeyyyv have you tried, in the send wallet, go to advanced and check

![image](https://user-images.githubusercontent.com/3020646/130582186-5ed15538-cf3a-4e22-a595-34d32a487903.png)

**Comment by pavlenex** [comment]
Yes issue reported on Trezor T as well @NicolasDorier.

**Comment by NicolasDorier** [comment]
I lost the cable, so I need to order a new one. Will take a bit of time to look at it

**Comment by gruve-p** [comment]
@pavlenex @mikeyyyv @dennisreimann 
On Trezor devices its mandatory to include non witness UTXO otherwise signing will always fail:
On the send wallet go to advanced settings and check `Always include non-witness UTXO if available`

**Comment by dennisreimann** [comment]
@gruve-p @NicolasDorier Yes, I can confirm that setting the "Always include non-witness UTXO" fixes the problem.

Can we circumvent this issue by detecting if it is a hardware wallet that needs this setting and include it later on, instead of relying on the user to know about it and dig it up from the "Advanced settings"? 

https://user-images.githubusercontent.com/886/130791384-4311bb03-51bd-4ae4-8285-685de0350f3d.MOV

**Comment by gruve-p** [comment]
Maybe best way is if a Trezor device is detected that the checkbox is automatically checked? Or maybe save user preferences when sending so user does not need to check it again?

P.S. This issue was reported to Nicolas on 12 April 2021 (via Mattermost) while porting BTCPayServer.Vault to GRSPayServer.Vault

**Comment by piliugin-anton** [comment]
Did you tried to report the problem to Trezor team?

**Comment by NicolasDorier** [comment]
@piliugin-anton they probably know already about it, but won't fix it. Ask them why their wallet requires the previous tx and not only non_witness_utxo.

**Comment by gruve-p** [comment]
cc @prusnak @matejcik

**Comment by prusnak** [comment]
> Ask them why their wallet requires the previous tx and not only non_witness_utxo.

We need previous TX to mitigate the design flaw of Bitcoin which makes it possible to steal user funds via really high transaction fee.

Good news is that the issue will be fixed with Taproot. When all inputs are Taproot then it is not required to provide previous TX to Trezor device.

**Comment by Monique1965** [comment]
Receiving wallet 13gaAfUXNrXcLQ4JLzYQR7r9uvnLve2Xns I'll Contribute  give 1/4 of this if y'all help me receive my pmt
